### PR TITLE
Issue 16904

### PR DESCRIPTION
**Pull request by erickgonzalez**

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/ContentletAPITest.java [hunk 1]
```diff
@@ -5943,4 +5943,50 @@ public void testCheckInContentletWithoutHost_shouldUseContentTypeHost()
         }
     }
 
+    @Test
+    public void testRemoveContentFromIndexMultilingualContent()
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/ContentletAPITest.java]: please add javadoc explaining what the test tests

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/ContentletAPITest.java [hunk 2]
```diff
@@ -5943,4 +5943,50 @@ public void testCheckInContentletWithoutHost_shouldUseContentTypeHost()
...
+            contentInSpanish.setLanguageId(spanishLanguage.getId());
+            contentInSpanish.setIndexPolicy(IndexPolicy.FORCE);
+            contentInSpanish = contentletDataGen.checkin(contentInSpanish);
+
+            assertEquals(2,
+                    contentletAPI.searchIndex("+identifier:"+contentInEnglish.getIdentifier(),-1,0,"",user,true).size());
+
+            contentletDataGen.archive(contentInSpanish);
+            contentletDataGen.delete(contentInSpanish);
+            Thread.sleep(1000);//give a sec so index refresh
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/ContentletAPITest.java]: should not be needed

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/ContentletAPITest.java [hunk 3]
```diff
@@ -5943,4 +5943,50 @@ public void testCheckInContentletWithoutHost_shouldUseContentTypeHost()
...
+                    .fields(ImmutableList
+                            .of(ImmutableTextField.builder().name("Title").variable("title")
+                                    .build()))
+                    .nextPersisted();
+
+            final ContentletDataGen contentletDataGen = new ContentletDataGen(contentType.id());
+            final Contentlet contentInEnglish = contentletDataGen.languageId(languageAPI.getDefaultLanguage().getId()).nextPersisted();
+            Contentlet contentInSpanish = contentletDataGen.checkout(contentInEnglish);
+            contentInSpanish.setLanguageId(spanishLanguage.getId());
+            contentInSpanish.setIndexPolicy(IndexPolicy.FORCE);
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/ContentletAPITest.java]: try with `WAIT_FOR`

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java [hunk 4]
```diff
@@ -1762,6 +1762,7 @@ private boolean destroyContentlets(List<Contentlet> contentlets, User user, bool
             }
             contentletsVersion.addAll(findAllVersions(APILocator.getIdentifierAPI().find(con.getIdentifier()), user,
                     respectFrontendRoles));
+            contentletsVersion.forEach(contentletLanguage -> contentletLanguage.setIndexPolicy(con.getIndexPolicy()));
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java]: nice

**Review by dsilvam — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Pull request merged**